Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp

```diff
@@ -3185,6 +3311,189 @@ auto B3IRGenerator::addCallRef(const TypeDefinition& originalSignature, Vector<E
     return emitIndirectCall(calleeInstance, calleeCode, signature, args, results);
 }
 
+auto B3IRGenerator::addReturnCall(uint32_t functionIndex, const TypeDefinition& signature, Vector<ExpressionType>& args) -> PartialResult
+{
+
+    const TypeIndex callerTypeIndex = m_info.internalFunctionTypeIndices[m_functionIndex];
```

Comment:
ditto on the code duplication. It will make it easier to review too.

Reply:
will do